xds/internal/xdsclient/xdsresource/filter_chain.go

```diff
@@ -194,10 +206,28 @@ type FilterChainManager struct {
 
 	def *FilterChain // Default filter chain, if specified.
 
+	// Slice of filter chains managed by this filter chain manager.
+	fcs []*FilterChain
+
 	// RouteConfigNames are the route configuration names which need to be
 	// dynamically queried for RDS Configuration for any FilterChains which
 	// specify to load RDS Configuration dynamically.
 	RouteConfigNames map[string]bool
+
+	// Persisted to gracefully close once filter chain manager no longer active.
+	conns []net.Conn
+}
+
+// AddConn adds the passed connection to the list of Conns in the filter chain
+// manager. Must not be called concurrently with Conns().
+func (fcm *FilterChainManager) AddConn(conn net.Conn) {
```

Comment:
Re: an earlier comment: can this be some other interface that includes a `Drain` method, instead of `net.Conn`, since this is not going to be able to accept `*connWrapper`?

Reply:
Added a type // DrainConn is an interface which specifies a Drain method.
type DrainConn interface {
	// Drain gracefully closes the Connection.
	Drain()
} in the xdsresource package.